scripts/postinstall.js

```diff
@@ -91,7 +128,23 @@ async function main() {
 
   console.info("Downloading", url);
   const resp = await fetch(url);
+
+  // checksum verification
+  resp.body
+    .pipe(hash) // Pipe to the hash calculator
+    .on("data", (data) => {
+      // Update the hash calculation while reading the data
+      hash.update(data);
+    })
+    .on("end", () => {
+      const calculatedChecksum = hash.digest("hex");
+      if (calculatedChecksum !== expectedChecksum) {
+        throw checksumError;
+      }
+    });
+
   resp.body.pipe(ungz).pipe(untar);
+
   await new Promise((resolve, reject) => {
     untar.on("error", reject);
     untar.on("end", () => resolve());
```

Comment:
Sorry it's not immediately obvious to me why there's an error. Perhaps you can try registering the callbacks in this promise instead, similar to what we've been doing for untar.